src/KDMqtt/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# This file is part of KDUtils.
+#
+# SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+# Author: Marco Thaller <marco.thaller@kdab.com>
+#
+# SPDX-License-Identifier: MIT
+#
+# Contact KDAB at <info@kdab.com> for commercial licensing options.
+#
+
+set(SOURCES mqtt.cpp)
+
+set(HEADERS mosquitto_wrapper.h mqtt.h)
+
+add_library(
+    KDMqtt
+    ${SOURCES} ${HEADERS}
+)
+
+add_library(
+    KDUtils::KDMqtt ALIAS KDMqtt
+)
+
+target_link_libraries(
+    KDMqtt
+    PUBLIC KDUtils::KDFoundation
+    PUBLIC Mosquitto::Mosquitto
+)
+
+target_include_directories(
+    KDMqtt
+    PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../..>
```

Comment:
this makes `mosquitto_wrapper.h` a public header. Shouldn't we rather make it an implementation detail and private?